// @revealjs/react/dist/index.mjs
import { Children as e, Fragment as t, cloneElement as n, createContext as r, isValidElement as i, useContext as a, useEffect as o, useLayoutEffect as s, useMemo as c, useRef as l, useState as u } from "react";
import d from "reveal.js";
import { Fragment as f, jsx as p, jsxs as m } from "react/jsx-runtime";
//#region src/reveal-context.ts
var h = r(null), ee = [];
function te(e, t) {
	if (e === t) return !1;
	if (!e || !t) return e !== t;
	let n = Object.keys(e), r = Object.keys(t);
	if (n.length !== r.length) return !0;
	for (let r of n) if (!(r in t) || e[r] !== t[r]) return !0;
	return !1;
}
function g(e, t) {
	e && (typeof e == "function" ? e(t) : e.current = t);
}
function _(e) {
	return e.tagName === "SECTION";
}
function v(e, t, n) {
	return Array.from(e.children).filter(_).map((e) => {
		let r = t.get(e);
		r === void 0 && (r = n.current++, t.set(e, r));
		let i = v(e, t, n);
		return i.length > 0 ? [r, i] : r;
	});
}
function y(e, t, n) {
	return e ? JSON.stringify(v(e, t, n)) : "[]";
}
function b({ config: e, plugins: t = ee, onReady: n, onSync: r, onSlideSync: i, onSlideChange: a, onSlideTransitionEnd: c, onFragmentShown: f, onFragmentHidden: m, onOverviewShown: _, onOverviewHidden: v, onPaused: b, onResumed: x, deckRef: S, className: C, style: w, children: ne }) {
	let T = l(null), E = l(null), D = l(null), [O, k] = u(null), re = l(t), A = l(!1), j = l(e), M = l(null), N = l(/* @__PURE__ */ new WeakMap()), P = l(1), F = l(!1), I = l(0);
	return o(() => {
		if (F.current = !0, I.current += 1, D.current) D.current.isReady() && k(D.current);
		else {
			let t = new d(T.current, {
				...e,
				plugins: re.current
			});
			j.current = e, D.current = t, t.initialize().then(() => {
				!F.current || D.current !== t || (k(t), n?.(t));
			});
		}
		return () => {
			F.current = !1;
			let e = D.current;
			if (!e) return;
			let t = ++I.current;
			Promise.resolve().then(() => {
				if (!(F.current || I.current !== t) && D.current === e) {
					try {
						e.destroy();
					} catch {}
					D.current === e && (D.current = null);
				}
			});
		};
	}, []), o(() => (g(S, O), () => g(S, null)), [S, O]), o(() => {
		if (!O) return;
		let e = [
			["sync", r],
			["slidesync", i],
			["slidechanged", a],
			["slidetransitionend", c],
			["fragmentshown", f],
			["fragmenthidden", m],
			["overviewshown", _],
			["overviewhidden", v],
			["paused", b],
			["resumed", x]
		].filter((e) => e[1] != null);
		for (let [t, n] of e) O.on(t, n);
		return () => {
			for (let [t, n] of e) O.off(t, n);
		};
	}, [
		O,
		r,
		i,
		a,
		c,
		f,
		m,
		_,
		v,
		b,
		x
	]), s(() => {
		!O || !D.current?.isReady() || te(j.current, e) && (A.current = !0, D.current.configure(e ?? {}), j.current = e);
	}, [O, e]), s(() => {
		let e = A.current;
		A.current = !1;
		let t = y(E.current, N.current, P);
		if (e) {
			M.current = t;
			return;
		}
		D.current?.isReady() && M.current !== t && (D.current.sync(), M.current = t);
	}), /* @__PURE__ */ p(h.Provider, {
		value: O,
		children: /* @__PURE__ */ p("div", {
			className: C ? `reveal ${C}` : "reveal",
			style: w,
			ref: T,
			children: /* @__PURE__ */ p("div", {
				className: "slides",
				ref: E,
				children: ne
			})
		})
	});
}
//#endregion
//#region src/utils/slide-attributes.ts
var x = {
	background: "data-background",
	backgroundImage: "data-background-image",
	backgroundVideo: "data-background-video",
	backgroundVideoLoop: "data-background-video-loop",
	backgroundVideoMuted: "data-background-video-muted",
	backgroundIframe: "data-background-iframe",
	backgroundColor: "data-background-color",
	backgroundGradient: "data-background-gradient",
	backgroundSize: "data-background-size",
	backgroundPosition: "data-background-position",
	backgroundRepeat: "data-background-repeat",
	backgroundOpacity: "data-background-opacity",
	backgroundTransition: "data-background-transition",
	visibility: "data-visibility",
	autoAnimate: "data-auto-animate",
	autoAnimateId: "data-auto-animate-id",
	autoAnimateRestart: "data-auto-animate-restart",
	autoAnimateUnmatched: "data-auto-animate-unmatched",
	autoAnimateEasing: "data-auto-animate-easing",
	autoAnimateDuration: "data-auto-animate-duration",
	autoAnimateDelay: "data-auto-animate-delay",
	transition: "data-transition",
	transitionSpeed: "data-transition-speed",
	autoSlide: "data-autoslide",
	notes: "data-notes",
	backgroundInteractive: "data-background-interactive",
	preload: "data-preload"
};
function S(e) {
	return JSON.stringify(Object.entries(e).filter(([e]) => e.startsWith("data-")).sort(([e], [t]) => e.localeCompare(t)));
}
function C(e, t) {
	let n = { ...e }, r = n;
	for (let [e, n] of Object.entries(x)) {
		if (r[n] !== void 0) continue;
		let i = t[e];
		if (i !== void 0) {
			if (i === !1) {
				e === "autoAnimateUnmatched" && (r[n] = "false");
				continue;
			}
			r[n] = typeof i == "boolean" ? "" : i;
		}
	}
	return n;
}
//#endregion
//#region src/components/slide.tsx
function w({ children: e, background: t, backgroundImage: n, backgroundVideo: r, backgroundVideoLoop: i, backgroundVideoMuted: o, backgroundIframe: c, backgroundColor: u, backgroundGradient: d, backgroundSize: f, backgroundPosition: m, backgroundRepeat: ee, backgroundOpacity: te, backgroundTransition: g, visibility: _, autoAnimate: v, autoAnimateId: y, autoAnimateRestart: b, autoAnimateUnmatched: x, autoAnimateEasing: w, autoAnimateDuration: ne, autoAnimateDelay: T, transition: E, transitionSpeed: D, autoSlide: O, notes: k, backgroundInteractive: re, preload: A, ...j }) {
	let M = a(h), N = l(null), P = l(null), F = l(null), I = C(j, {
		background: t,
		backgroundImage: n,
		backgroundVideo: r,
		backgroundVideoLoop: i,
		backgroundVideoMuted: o,
		backgroundIframe: c,
		backgroundColor: u,
		backgroundGradient: d,
		backgroundSize: f,
		backgroundPosition: m,
		backgroundRepeat: ee,
		backgroundOpacity: te,
		backgroundTransition: g,
		visibility: _,
		autoAnimate: v,
		autoAnimateId: y,
		autoAnimateRestart: b,
		autoAnimateUnmatched: x,
		autoAnimateEasing: w,
		autoAnimateDuration: ne,
		autoAnimateDelay: T,
		transition: E,
		transitionSpeed: D,
		autoSlide: O,
		notes: k,
		backgroundInteractive: re,
		preload: A
	}), L = S(I);
	return s(() => {
		let e = N.current;
		if (!M || !e || typeof M.syncSlide != "function") return;
		if (P.current !== M) {
			P.current = M, F.current = L;
			return;
		}
		if (L === "[]") return;
		let t = P.current === M, n = F.current === L;
		t && n || (M.syncSlide(e), P.current = M, F.current = L);
	}, [M, L]), /* @__PURE__ */ p("section", {
		ref: N,
		...I,
		children: e
	});
}
//#endregion
//#region src/components/stack.tsx
function ne({ className: e, style: t, children: n }) {
	return /* @__PURE__ */ p("section", {
		className: e,
		style: t,
		children: n
	});
}
//#endregion
//#region ../node_modules/marked/lib/marked.esm.js
function T() {
	return {
		async: !1,
		breaks: !1,
		extensions: null,
		gfm: !0,
		hooks: null,
		pedantic: !1,
		renderer: null,
		silent: !1,
		tokenizer: null,
		walkTokens: null
	};
}
var E = T();
function D(e) {
	E = e;
}
var O = { exec: () => null };
function k(e, t = "") {
	let n = typeof e == "string" ? e : e.source, r = {
		replace: (e, t) => {
			let i = typeof t == "string" ? t : t.source;
			return i = i.replace(A.caret, "$1"), n = n.replace(e, i), r;
		},
		getRegex: () => new RegExp(n, t)
	};
	return r;
}
var re = (() => {
	try {
		return !0;
	} catch {
		return !1;
	}
})(), A = {
	codeRemoveIndent: /^(?: {1,4}| {0,3}\t)/gm,
	outputLinkReplace: /\\([\[\]])/g,
	indentCodeCompensation: /^(\s+)(?:```)/,
	beginningSpace: /^\s+/,
	endingHash: /#$/,
	startingSpaceChar: /^ /,
	endingSpaceChar: / $/,
	nonSpaceChar: /[^ ]/,
	newLineCharGlobal: /\n/g,
	tabCharGlobal: /\t/g,
	multipleSpaceGlobal: /\s+/g,
	blankLine: /^[ \t]*$/,
	doubleBlankLine: /\n[ \t]*\n[ \t]*$/,
	blockquoteStart: /^ {0,3}>/,
	blockquoteSetextReplace: /\n {0,3}((?:=+|-+) *)(?=\n|$)/g,
	blockquoteSetextReplace2: /^ {0,3}>[ \t]?/gm,
	listReplaceNesting: /^ {1,4}(?=( {4})*[^ ])/g,
	listIsTask: /^\[[ xX]\] +\S/,
	listReplaceTask: /^\[[ xX]\] +/,
	listTaskCheckbox: /\[[ xX]\]/,
	anyLine: /\n.*\n/,
	hrefBrackets: /^<(.*)>$/,
	tableDelimiter: /[:|]/,
	tableAlignChars: /^\||\| *$/g,
	tableRowBlankLine: /\n[ \t]*$/,
	tableAlignRight: /^ *-+: *$/,
	tableAlignCenter: /^ *:-+: *$/,
	tableAlignLeft: /^ *:-+ *$/,
	startATag: /^<a /i,
	endATag: /^<\/a>/i,
	startPreScriptTag: /^<(pre|code|kbd|script)(\s|>)/i,
	endPreScriptTag: /^<\/(pre|code|kbd|script)(\s|>)/i,
	startAngleBracket: /^</,
	endAngleBracket: />$/,
	pedanticHrefTitle: /^([^'"]*[^\s])\s+(['"])(.*)\2/,
	unicodeAlphaNumeric: /[\p{L}\p{N}]/u,
	escapeTest: /[&<>"']/,
	escapeReplace: /[&<>"']/g,
	escapeTestNoEncode: /[<>"']|&(?!(#\d{1,7}|#[Xx][a-fA-F0-9]{1,6}|\w+);)/,
	escapeReplaceNoEncode: /[<>"']|&(?!(#\d{1,7}|#[Xx][a-fA-F0-9]{1,6}|\w+);)/g,
	caret: /(^|[^\[])\^/g,
	percentDecode: /%25/g,
	findPipe: /\|/g,
	splitPipe: / \|/,
	slashPipe: /\\\|/g,
	carriageReturn: /\r\n|\r/g,
	spaceLine: /^ +$/gm,
	notSpaceStart: /^\S*/,
	endingNewline: /\n$/,
	listItemRegex: (e) => RegExp(`^( {0,3}${e})((?:[	 ][^\\n]*)?(?:\\n|$))`),
	nextBulletRegex: (e) => RegExp(`^ {0,${Math.min(3, e - 1)}}(?:[*+-]|\\d{1,9}[.)])((?:[ 	][^\\n]*)?(?:\\n|$))`),
	hrRegex: (e) => RegExp(`^ {0,${Math.min(3, e - 1)}}((?:- *){3,}|(?:_ *){3,}|(?:\\* *){3,})(?:\\n+|$)`),
	fencesBeginRegex: (e) => RegExp(`^ {0,${Math.min(3, e - 1)}}(?:\`\`\`|~~~)`),
	headingBeginRegex: (e) => RegExp(`^ {0,${Math.min(3, e - 1)}}#`),
	htmlBeginRegex: (e) => RegExp(`^ {0,${Math.min(3, e - 1)}}<(?:[a-z].*>|!--)`, "i"),
	blockquoteBeginRegex: (e) => RegExp(`^ {0,${Math.min(3, e - 1)}}>`)
}, j = /^(?:[ \t]*(?:\n|$))+/, M = /^((?: {4}| {0,3}\t)[^\n]+(?:\n(?:[ \t]*(?:\n|$))*)?)+/, N = /^ {0,3}(`{3,}(?=[^`\n]*(?:\n|$))|~{3,})([^\n]*)(?:\n|$)(?:|([\s\S]*?)(?:\n|$))(?: {0,3}\1[~`]* *(?=\n|$)|$)/, P = /^ {0,3}((?:-[\t ]*){3,}|(?:_[ \t]*){3,}|(?:\*[ \t]*){3,})(?:\n+|$)/, F = /^ {0,3}(#{1,6})(?=\s|$)(.*)(?:\n+|$)/, I = / {0,3}(?:[*+-]|\d{1,9}[.)])/, L = /^(?!bull |blockCode|fences|blockquote|heading|html|table)((?:.|\n(?!\s*?\n|bull |blockCode|fences|blockquote|heading|html|table))+?)\n {0,3}(=+|-+) *(?:\n+|$)/, ie = k(L).replace(/bull/g, I).replace(/blockCode/g, /(?: {4}| {0,3}\t)/).replace(/fences/g, / {0,3}(?:`{3,}|~{3,})/).replace(/blockquote/g, / {0,3}>/).replace(/heading/g, / {0,3}#{1,6}/).replace(/html/g, / {0,3}<[^\n>]+>\n/).replace(/\|table/g, "").getRegex(), ae = k(L).replace(/bull/g, I).replace(/blockCode/g, /(?: {4}| {0,3}\t)/).replace(/fences/g, / {0,3}(?:`{3,}|~{3,})/).replace(/blockquote/g, / {0,3}>/).replace(/heading/g, / {0,3}#{1,6}/).replace(/html/g, / {0,3}<[^\n>]+>\n/).replace(/table/g, / {0,3}\|?(?:[:\- ]*\|)+[\:\- ]*\n/).getRegex(), R = /^([^\n]+(?:\n(?!hr|heading|lheading|blockquote|fences|list|html|table| +\n)[^\n]+)*)/, z = /^[^\n]+/, B = /(?!\s*\])(?:\\[\s\S]|[^\[\]\\])+/, V = k(/^ {0,3}\[(label)\]: *(?:\n[ \t]*)?([^<\s][^\s]*|<.*?>)(?:(?: +(?:\n[ \t]*)?| *\n[ \t]*)(title))? *(?:\n+|$)/).replace("label", B).replace("title", /(?:"(?:\\"?|[^"\\])*"|'[^'\n]*(?:\n[^'\n]+)*\n?'|\([^()]*\))/).getRegex(), H = k(/^(bull)([ \t][^\n]+?)?(?:\n|$)/).replace(/bull/g, I).getRegex(), U = "address|article|aside|base|basefont|blockquote|body|caption|center|col|colgroup|dd|details|dialog|dir|div|dl|dt|fieldset|figcaption|figure|footer|form|frame|frameset|h[1-6]|head|header|hr|html|iframe|legend|li|link|main|menu|menuitem|meta|nav|noframes|ol|optgroup|option|p|param|search|section|summary|table|tbody|td|tfoot|th|thead|title|tr|track|ul", W = /<!--(?:-?>|[\s\S]*?(?:-->|$))/, oe = k("^ {0,3}(?:<(script|pre|style|textarea)[\\s>][\\s\\S]*?(?:</\\1>[^\\n]*\\n+|$)|comment[^\\n]*(\\n+|$)|<\\?[\\s\\S]*?(?:\\?>\\n*|$)|<![A-Z][\\s\\S]*?(?:>\\n*|$)|<!\\[CDATA\\[[\\s\\S]*?(?:\\]\\]>\\n*|$)|</?(tag)(?: +|\\n|/?>)[\\s\\S]*?(?:(?:\\n[ 	]*)+\\n|$)|<(?!script|pre|style|textarea)([a-z][\\w-]*)(?:attribute)*? */?>(?=[ \\t]*(?:\\n|$))[\\s\\S]*?(?:(?:\\n[ 	]*)+\\n|$)|</(?!script|pre|style|textarea)[a-z][\\w-]*\\s*>(?=[ \\t]*(?:\\n|$))[\\s\\S]*?(?:(?:\\n[ 	]*)+\\n|$))", "i").replace("comment", W).replace("tag", U).replace("attribute", / +[a-zA-Z:_][\w.:-]*(?: *= *"[^"\n]*"| *= *'[^'\n]*'| *= *[^\s"'=<>`]+)?/).getRegex(), se = k(R).replace("hr", P).replace("heading", " {0,3}#{1,6}(?:\\s|$)").replace("|lheading", "").replace("|table", "").replace("blockquote", " {0,3}>").replace("fences", " {0,3}(?:`{3,}(?=[^`\\n]*\\n)|~{3,})[^\\n]*\\n").replace("list", " {0,3}(?:[*+-]|1[.)])[ \\t]").replace("html", "</?(?:tag)(?: +|\\n|/?>)|<(?:script|pre|style|textarea|!--)").replace("tag", U).getRegex(), ce = {
	blockquote: k(/^( {0,3}> ?(paragraph|[^\n]*)(?:\n|$))+/).replace("paragraph", se).getRegex(),
	code: M,
	def: V,
	fences: N,
	heading: F,
	hr: P,
	html: oe,
	lheading: ie,
	list: H,
	newline: j,
	paragraph: se,
	table: O,
	text: z
}, le = k("^ *([^\\n ].*)\\n {0,3}((?:\\| *)?:?-+:? *(?:\\| *:?-+:? *)*(?:\\| *)?)(?:\\n((?:(?! *\\n|hr|heading|blockquote|code|fences|list|html).*(?:\\n|$))*)\\n*|$)").replace("hr", P).replace("heading", " {0,3}#{1,6}(?:\\s|$)").replace("blockquote", " {0,3}>").replace("code", "(?: {4}| {0,3}	)[^\\n]").replace("fences", " {0,3}(?:`{3,}(?=[^`\\n]*\\n)|~{3,})[^\\n]*\\n").replace("list", " {0,3}(?:[*+-]|1[.)])[ \\t]").replace("html", "</?(?:tag)(?: +|\\n|/?>)|<(?:script|pre|style|textarea|!--)").replace("tag", U).getRegex(), ue = {
	...ce,
	lheading: ae,
	table: le,
	paragraph: k(R).replace("hr", P).replace("heading", " {0,3}#{1,6}(?:\\s|$)").replace("|lheading", "").replace("table", le).replace("blockquote", " {0,3}>").replace("fences", " {0,3}(?:`{3,}(?=[^`\\n]*\\n)|~{3,})[^\\n]*\\n").replace("list", " {0,3}(?:[*+-]|1[.)])[ \\t]").replace("html", "</?(?:tag)(?: +|\\n|/?>)|<(?:script|pre|style|textarea|!--)").replace("tag", U).getRegex()
}, de = {
	...ce,
	html: k("^ *(?:comment *(?:\\n|\\s*$)|<(tag)[\\s\\S]+?</\\1> *(?:\\n{2,}|\\s*$)|<tag(?:\"[^\"]*\"|'[^']*'|\\s[^'\"/>\\s]*)*?/?> *(?:\\n{2,}|\\s*$))").replace("comment", W).replace(/tag/g, "(?!(?:a|em|strong|small|s|cite|q|dfn|abbr|data|time|code|var|samp|kbd|sub|sup|i|b|u|mark|ruby|rt|rp|bdi|bdo|span|br|wbr|ins|del|img)\\b)\\w+(?!:|[^\\w\\s@]*@)\\b").getRegex(),
	def: /^ *\[([^\]]+)\]: *<?([^\s>]+)>?(?: +(["(][^\n]+[")]))? *(?:\n+|$)/,
	heading: /^(#{1,6})(.*)(?:\n+|$)/,
	fences: O,
	lheading: /^(.+?)\n {0,3}(=+|-+) *(?:\n+|$)/,
	paragraph: k(R).replace("hr", P).replace("heading", " *#{1,6} *[^\n]").replace("lheading", ie).replace("|table", "").replace("blockquote", " {0,3}>").replace("|fences", "").replace("|list", "").replace("|html", "").replace("|tag", "").getRegex()
}, fe = /^\\([!"#$%&'()*+,\-./:;<=>?@\[\]\\^_`{|}~])/, pe = /^(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)/, me = /^( {2,}|\\)\n(?!\s*$)/, he = /^(`+|[^`])(?:(?= {2,}\n)|[\s\S]*?(?:(?=[\\<!\[`*_]|\b_|$)|[^ ](?= {2,}\n)))/, G = /[\p{P}\p{S}]/u, ge = /[\s\p{P}\p{S}]/u, _e = /[^\s\p{P}\p{S}]/u, ve = k(/^((?![*_])punctSpace)/, "u").replace(/punctSpace/g, ge).getRegex(), ye = /(?!~)[\p{P}\p{S}]/u, be = /(?!~)[\s\p{P}\p{S}]/u, xe = /(?:[^\s\p{P}\p{S}]|~)/u, Se = k(/link|precode-code|html/, "g").replace("link", /\[(?:[^\[\]`]|(?<a>`+)[^`]+\k<a>(?!`))*?\]\((?:\\[\s\S]|[^\\\(\)]|\((?:\\[\s\S]|[^\\\(\)])*\))*\)/).replace("precode-", re ? "(?<!`)()" : "(^^|[^`])").replace("code", /(?<b>`+)[^`]+\k<b>(?!`)/).replace("html", /<(?! )[^<>]*?>/).getRegex(), Ce = /^(?:\*+(?:((?!\*)punct)|([^\s*]))?)|^_+(?:((?!_)punct)|([^\s_]))?/, we = k(Ce, "u").replace(/punct/g, G).getRegex(), Te = k(Ce, "u").replace(/punct/g, ye).getRegex(), Ee = "^[^_*]*?__[^_*]*?\\*[^_*]*?(?=__)|[^*]+(?=[^*])|(?!\\*)punct(\\*+)(?=[\\s]|$)|notPunctSpace(\\*+)(?!\\*)(?=punctSpace|$)|(?!\\*)punctSpace(\\*+)(?=notPunctSpace)|[\\s](\\*+)(?!\\*)(?=punct)|(?!\\*)punct(\\*+)(?!\\*)(?=punct)|notPunctSpace(\\*+)(?=notPunctSpace)", De = k(Ee, "gu").replace(/notPunctSpace/g, _e).replace(/punctSpace/g, ge).replace(/punct/g, G).getRegex(), Oe = k(Ee, "gu").replace(/notPunctSpace/g, xe).replace(/punctSpace/g, be).replace(/punct/g, ye).getRegex(), ke = k("^[^_*]*?\\*\\*[^_*]*?_[^_*]*?(?=\\*\\*)|[^_]+(?=[^_])|(?!_)punct(_+)(?=[\\s]|$)|notPunctSpace(_+)(?!_)(?=punctSpace|$)|(?!_)punctSpace(_+)(?=notPunctSpace)|[\\s](_+)(?!_)(?=punct)|(?!_)punct(_+)(?!_)(?=punct)", "gu").replace(/notPunctSpace/g, _e).replace(/punctSpace/g, ge).replace(/punct/g, G).getRegex(), Ae = k(/^~~?(?:((?!~)punct)|[^\s~])/, "u").replace(/punct/g, G).getRegex(), je = k("^[^~]+(?=[^~])|(?!~)punct(~~?)(?=[\\s]|$)|notPunctSpace(~~?)(?!~)(?=punctSpace|$)|(?!~)punctSpace(~~?)(?=notPunctSpace)|[\\s](~~?)(?!~)(?=punct)|(?!~)punct(~~?)(?!~)(?=punct)|notPunctSpace(~~?)(?=notPunctSpace)", "gu").replace(/notPunctSpace/g, _e).replace(/punctSpace/g, ge).replace(/punct/g, G).getRegex(), Me = k(/\\(punct)/, "gu").replace(/punct/g, G).getRegex(), Ne = k(/^<(scheme:[^\s\x00-\x1f<>]*|email)>/).replace("scheme", /[a-zA-Z][a-zA-Z0-9+.-]{1,31}/).replace("email", /[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+(@)[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+(?![-_])/).getRegex(), Pe = k(W).replace("(?:-->|$)", "-->").getRegex(), Fe = k("^comment|^</[a-zA-Z][\\w:-]*\\s*>|^<[a-zA-Z][\\w-]*(?:attribute)*?\\s*/?>|^<\\?[\\s\\S]*?\\?>|^<![a-zA-Z]+\\s[\\s\\S]*?>|^<!\\[CDATA\\[[\\s\\S]*?\\]\\]>").replace("comment", Pe).replace("attribute", /\s+[a-zA-Z:_][\w.:-]*(?:\s*=\s*"[^"]*"|\s*=\s*'[^']*'|\s*=\s*[^\s"'=<>`]+)?/).getRegex(), Ie = /(?:\[(?:\\[\s\S]|[^\[\]\\])*\]|\\[\s\S]|`+(?!`)[^`]*?`+(?!`)|``+(?=\])|[^\[\]\\`])*?/, Le = k(/^!?\[(label)\]\(\s*(href)(?:(?:[ \t]+(?:\n[ \t]*)?|\n[ \t]*)(title))?\s*\)/).replace("label", Ie).replace("href", /<(?:\\.|[^\n<>\\])+>|[^ \t\n\x00-\x1f]*/).replace("title", /"(?:\\"?|[^"\\])*"|'(?:\\'?|[^'\\])*'|\((?:\\\)?|[^)\\])*\)/).getRegex(), Re = k(/^!?\[(label)\]\[(ref)\]/).replace("label", Ie).replace("ref", B).getRegex(), ze = k(/^!?\[(ref)\](?:\[\])?/).replace("ref", B).getRegex(), Be = k("reflink|nolink(?!\\()", "g").replace("reflink", Re).replace("nolink", ze).getRegex(), Ve = /[hH][tT][tT][pP][sS]?|[fF][tT][pP]/, He = {
	_backpedal: O,
	anyPunctuation: Me,
	autolink: Ne,
	blockSkip: Se,
	br: me,
	code: pe,
	del: O,
	delLDelim: O,
	delRDelim: O,
	emStrongLDelim: we,
	emStrongRDelimAst: De,
	emStrongRDelimUnd: ke,
	escape: fe,
	link: Le,
	nolink: ze,
	punctuation: ve,
	reflink: Re,
	reflinkSearch: Be,
	tag: Fe,
	text: he,
	url: O
}, Ue = {
	...He,
	link: k(/^!?\[(label)\]\((.*?)\)/).replace("label", Ie).getRegex(),
	reflink: k(/^!?\[(label)\]\s*\[([^\]]*)\]/).replace("label", Ie).getRegex()
}, We = {
	...He,
	emStrongRDelimAst: Oe,
	emStrongLDelim: Te,
	delLDelim: Ae,
	delRDelim: je,
	url: k(/^((?:protocol):\/\/|www\.)(?:[a-zA-Z0-9\-]+\.?)+[^\s<]*|^email/).replace("protocol", Ve).replace("email", /[A-Za-z0-9._+-]+(@)[a-zA-Z0-9-_]+(?:\.[a-zA-Z0-9-_]*[a-zA-Z0-9])+(?![-_])/).getRegex(),
	_backpedal: /(?:[^?!.,:;*_'"~()&]+|\([^)]*\)|&(?![a-zA-Z0-9]+;$)|[?!.,:;*_'"~)]+(?!$))+/,
	del: /^(~~?)(?=[^\s~])((?:\\[\s\S]|[^\\])*?(?:\\[\s\S]|[^\s~\\]))\1(?=[^~]|$)/,
	text: k(/^([`~]+|[^`~])(?:(?= {2,}\n)|(?=[a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-]+@)|[\s\S]*?(?:(?=[\\<!\[`*~_]|\b_|protocol:\/\/|www\.|$)|[^ ](?= {2,}\n)|[^a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-](?=[a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-]+@)))/).replace("protocol", Ve).getRegex()
}, Ge = {
	...We,
	br: k(me).replace("{2,}", "*").getRegex(),
	text: k(We.text).replace("\\b_", "\\b_| {2,}\\n").replace(/\{2,\}/g, "*").getRegex()
}, Ke = {
	normal: ce,
	gfm: ue,
	pedantic: de
}, K = {
	normal: He,
	gfm: We,
	breaks: Ge,
	pedantic: Ue
}, qe = {
	"&": "&amp;",
	"<": "&lt;",
	">": "&gt;",
	"\"": "&quot;",
	"'": "&#39;"
}, Je = (e) => qe[e];
function q(e, t) {
	if (t) {
		if (A.escapeTest.test(e)) return e.replace(A.escapeReplace, Je);
	} else if (A.escapeTestNoEncode.test(e)) return e.replace(A.escapeReplaceNoEncode, Je);
	return e;
}
function Ye(e) {
	try {
		e = encodeURI(e).replace(A.percentDecode, "%");
	} catch {
		return null;
	}
	return e;
}
function Xe(e, t) {
	let n = e.replace(A.findPipe, (e, t, n) => {
		let r = !1, i = t;
		for (; --i >= 0 && n[i] === "\\";) r = !r;
		return r ? "|" : " |";
	}).split(A.splitPipe), r = 0;
	if (n[0].trim() || n.shift(), n.length > 0 && !n.at(-1)?.trim() && n.pop(), t) if (n.length > t) n.splice(t);
	else for (; n.length < t;) n.push("");
	for (; r < n.length; r++) n[r] = n[r].trim().replace(A.slashPipe, "|");
	return n;
}
function J(e, t, n) {
	let r = e.length;
	if (r === 0) return "";
	let i = 0;
	for (; i < r;) {
		let a = e.charAt(r - i - 1);
		if (a === t && !n) i++;
		else if (a !== t && n) i++;
		else break;
	}
	return e.slice(0, r - i);
}
function Ze(e, t) {
	if (e.indexOf(t[1]) === -1) return -1;
	let n = 0;
	for (let r = 0; r < e.length; r++) if (e[r] === "\\") r++;
	else if (e[r] === t[0]) n++;
	else if (e[r] === t[1] && (n--, n < 0)) return r;
	return n > 0 ? -2 : -1;
}
function Qe(e, t = 0) {
	let n = t, r = "";
	for (let t of e) if (t === "	") {
		let e = 4 - n % 4;
		r += " ".repeat(e), n += e;
	} else r += t, n++;
	return r;
}
function $e(e, t, n, r, i) {
	let a = t.href, o = t.title || null, s = e[1].replace(i.other.outputLinkReplace, "$1");
	r.state.inLink = !0;
	let c = {
		type: e[0].charAt(0) === "!" ? "image" : "link",
		raw: n,
		href: a,
		title: o,
		text: s,
		tokens: r.inlineTokens(s)
	};
	return r.state.inLink = !1, c;
}
function et(e, t, n) {
	let r = e.match(n.other.indentCodeCompensation);
	if (r === null) return t;
	let i = r[1];
	return t.split("\n").map((e) => {
		let t = e.match(n.other.beginningSpace);
		if (t === null) return e;
		let [r] = t;
		return r.length >= i.length ? e.slice(i.length) : e;
	}).join("\n");
}
var tt = class {
	options;
	rules;
	lexer;
	constructor(e) {
		this.options = e || E;
	}
	space(e) {
		let t = this.rules.block.newline.exec(e);
		if (t && t[0].length > 0) return {
			type: "space",
			raw: t[0]
		};
	}
	code(e) {
		let t = this.rules.block.code.exec(e);
		if (t) {
			let e = t[0].replace(this.rules.other.codeRemoveIndent, "");
			return {
				type: "code",
				raw: t[0],
				codeBlockStyle: "indented",
				text: this.options.pedantic ? e : J(e, "\n")
			};
		}
	}
	fences(e) {
		let t = this.rules.block.fences.exec(e);
		if (t) {
			let e = t[0], n = et(e, t[3] || "", this.rules);
			return {
				type: "code",
				raw: e,
				lang: t[2] ? t[2].trim().replace(this.rules.inline.anyPunctuation, "$1") : t[2],
				text: n
			};
		}
	}
	heading(e) {
		let t = this.rules.block.heading.exec(e);
		if (t) {
			let e = t[2].trim();
			if (this.rules.other.endingHash.test(e)) {
				let t = J(e, "#");
				(this.options.pedantic || !t || this.rules.other.endingSpaceChar.test(t)) && (e = t.trim());
			}
			return {
				type: "heading",
				raw: t[0],
				depth: t[1].length,
				text: e,
				tokens: this.lexer.inline(e)
			};
		}
	}
	hr(e) {
		let t = this.rules.block.hr.exec(e);
		if (t) return {
			type: "hr",
			raw: J(t[0], "\n")
		};
	}
	blockquote(e) {
		let t = this.rules.block.blockquote.exec(e);
		if (t) {
			let e = J(t[0], "\n").split("\n"), n = "", r = "", i = [];
			for (; e.length > 0;) {
				let t = !1, a = [], o;
				for (o = 0; o < e.length; o++) if (this.rules.other.blockquoteStart.test(e[o])) a.push(e[o]), t = !0;
				else if (!t) a.push(e[o]);
				else break;
				e = e.slice(o);
				let s = a.join("\n"), c = s.replace(this.rules.other.blockquoteSetextReplace, "\n    $1").replace(this.rules.other.blockquoteSetextReplace2, "");
				n = n ? `${n}
${s}` : s, r = r ? `${r}
${c}` : c;
				let l = this.lexer.state.top;
				if (this.lexer.state.top = !0, this.lexer.blockTokens(c, i, !0), this.lexer.state.top = l, e.length === 0) break;
				let u = i.at(-1);
				if (u?.type === "code") break;
				if (u?.type === "blockquote") {
					let t = u, a = t.raw + "\n" + e.join("\n"), o = this.blockquote(a);
					i[i.length - 1] = o, n = n.substring(0, n.length - t.raw.length) + o.raw, r = r.substring(0, r.length - t.text.length) + o.text;
					break;
				} else if (u?.type === "list") {
					let t = u, a = t.raw + "\n" + e.join("\n"), o = this.list(a);
					i[i.length - 1] = o, n = n.substring(0, n.length - u.raw.length) + o.raw, r = r.substring(0, r.length - t.raw.length) + o.raw, e = a.substring(i.at(-1).raw.length).split("\n");
					continue;
				}
			}
			return {
				type: "blockquote",
				raw: n,
				tokens: i,
				text: r
			};
		}
	}
	list(e) {
		let t = this.rules.block.list.exec(e);
		if (t) {
			let n = t[1].trim(), r = n.length > 1, i = {
				type: "list",
				raw: "",
				ordered: r,
				start: r ? +n.slice(0, -1) : "",
				loose: !1,
				items: []
			};
			n = r ? `\\d{1,9}\\${n.slice(-1)}` : `\\${n}`, this.options.pedantic && (n = r ? n : "[*+-]");
			let a = this.rules.other.listItemRegex(n), o = !1;
			for (; e;) {
				let n = !1, r = "", s = "";
				if (!(t = a.exec(e)) || this.rules.block.hr.test(e)) break;
				r = t[0], e = e.substring(r.length);
				let c = Qe(t[2].split("\n", 1)[0], t[1].length), l = e.split("\n", 1)[0], u = !c.trim(), d = 0;
				if (this.options.pedantic ? (d = 2, s = c.trimStart()) : u ? d = t[1].length + 1 : (d = c.search(this.rules.other.nonSpaceChar), d = d > 4 ? 1 : d, s = c.slice(d), d += t[1].length), u && this.rules.other.blankLine.test(l) && (r += l + "\n", e = e.substring(l.length + 1), n = !0), !n) {
					let t = this.rules.other.nextBulletRegex(d), n = this.rules.other.hrRegex(d), i = this.rules.other.fencesBeginRegex(d), a = this.rules.other.headingBeginRegex(d), o = this.rules.other.htmlBeginRegex(d), f = this.rules.other.blockquoteBeginRegex(d);
					for (; e;) {
						let p = e.split("\n", 1)[0], m;
						if (l = p, this.options.pedantic ? (l = l.replace(this.rules.other.listReplaceNesting, "  "), m = l) : m = l.replace(this.rules.other.tabCharGlobal, "    "), i.test(l) || a.test(l) || o.test(l) || f.test(l) || t.test(l) || n.test(l)) break;
						if (m.search(this.rules.other.nonSpaceChar) >= d || !l.trim()) s += "\n" + m.slice(d);
						else {
							if (u || c.replace(this.rules.other.tabCharGlobal, "    ").search(this.rules.other.nonSpaceChar) >= 4 || i.test(c) || a.test(c) || n.test(c)) break;
							s += "\n" + l;
						}
						u = !l.trim(), r += p + "\n", e = e.substring(p.length + 1), c = m.slice(d);
					}
				}
				i.loose || (o ? i.loose = !0 : this.rules.other.doubleBlankLine.test(r) && (o = !0)), i.items.push({
					type: "list_item",
					raw: r,
					task: !!this.options.gfm && this.rules.other.listIsTask.test(s),
					loose: !1,
					text: s,
					tokens: []
				}), i.raw += r;
			}
			let s = i.items.at(-1);
			if (s) s.raw = s.raw.trimEnd(), s.text = s.text.trimEnd();
			else return;
			i.raw = i.raw.trimEnd();
			for (let e of i.items) {
				if (this.lexer.state.top = !1, e.tokens = this.lexer.blockTokens(e.text, []), e.task) {
					if (e.text = e.text.replace(this.rules.other.listReplaceTask, ""), e.tokens[0]?.type === "text" || e.tokens[0]?.type === "paragraph") {
						e.tokens[0].raw = e.tokens[0].raw.replace(this.rules.other.listReplaceTask, ""), e.tokens[0].text = e.tokens[0].text.replace(this.rules.other.listReplaceTask, "");
						for (let e = this.lexer.inlineQueue.length - 1; e >= 0; e--) if (this.rules.other.listIsTask.test(this.lexer.inlineQueue[e].src)) {
							this.lexer.inlineQueue[e].src = this.lexer.inlineQueue[e].src.replace(this.rules.other.listReplaceTask, "");
							break;
						}
					}
					let t = this.rules.other.listTaskCheckbox.exec(e.raw);
					if (t) {
						let n = {
							type: "checkbox",
							raw: t[0] + " ",
							checked: t[0] !== "[ ]"
						};
						e.checked = n.checked, i.loose ? e.tokens[0] && ["paragraph", "text"].includes(e.tokens[0].type) && "tokens" in e.tokens[0] && e.tokens[0].tokens ? (e.tokens[0].raw = n.raw + e.tokens[0].raw, e.tokens[0].text = n.raw + e.tokens[0].text, e.tokens[0].tokens.unshift(n)) : e.tokens.unshift({
							type: "paragraph",
							raw: n.raw,
							text: n.raw,
							tokens: [n]
						}) : e.tokens.unshift(n);
					}
				}
				if (!i.loose) {
					let t = e.tokens.filter((e) => e.type === "space");
					i.loose = t.length > 0 && t.some((e) => this.rules.other.anyLine.test(e.raw));
				}
			}
			if (i.loose) for (let e of i.items) {
				e.loose = !0;
				for (let t of e.tokens) t.type === "text" && (t.type = "paragraph");
			}
			return i;
		}
	}
	html(e) {
		let t = this.rules.block.html.exec(e);
		if (t) return {
			type: "html",
			block: !0,
			raw: t[0],
			pre: t[1] === "pre" || t[1] === "script" || t[1] === "style",
			text: t[0]
		};
	}
	def(e) {
		let t = this.rules.block.def.exec(e);
		if (t) {
			let e = t[1].toLowerCase().replace(this.rules.other.multipleSpaceGlobal, " "), n = t[2] ? t[2].replace(this.rules.other.hrefBrackets, "$1").replace(this.rules.inline.anyPunctuation, "$1") : "", r = t[3] ? t[3].substring(1, t[3].length - 1).replace(this.rules.inline.anyPunctuation, "$1") : t[3];
			return {
				type: "def",
				tag: e,
				raw: t[0],
				href: n,
				title: r
			};
		}
	}
	table(e) {
		let t = this.rules.block.table.exec(e);
		if (!t || !this.rules.other.tableDelimiter.test(t[2])) return;
		let n = Xe(t[1]), r = t[2].replace(this.rules.other.tableAlignChars, "").split("|"), i = t[3]?.trim() ? t[3].replace(this.rules.other.tableRowBlankLine, "").split("\n") : [], a = {
			type: "table",
			raw: t[0],
			header: [],
			align: [],
			rows: []
		};
		if (n.length === r.length) {
			for (let e of r) this.rules.other.tableAlignRight.test(e) ? a.align.push("right") : this.rules.other.tableAlignCenter.test(e) ? a.align.push("center") : this.rules.other.tableAlignLeft.test(e) ? a.align.push("left") : a.align.push(null);
			for (let e = 0; e < n.length; e++) a.header.push({
				text: n[e],
				tokens: this.lexer.inline(n[e]),
				header: !0,
				align: a.align[e]
			});
			for (let e of i) a.rows.push(Xe(e, a.header.length).map((e, t) => ({
				text: e,
				tokens: this.lexer.inline(e),
				header: !1,
				align: a.align[t]
			})));
			return a;
		}
	}
	lheading(e) {
		let t = this.rules.block.lheading.exec(e);
		if (t) {
			let e = t[1].trim();
			return {
				type: "heading",
				raw: t[0],
				depth: t[2].charAt(0) === "=" ? 1 : 2,
				text: e,
				tokens: this.lexer.inline(e)
			};
		}
	}
	paragraph(e) {
		let t = this.rules.block.paragraph.exec(e);
		if (t) {
			let e = t[1].charAt(t[1].length - 1) === "\n" ? t[1].slice(0, -1) : t[1];
			return {
				type: "paragraph",
				raw: t[0],
				text: e,
				tokens: this.lexer.inline(e)
			};
		}
	}
	text(e) {
		let t = this.rules.block.text.exec(e);
		if (t) return {
			type: "text",
			raw: t[0],
			text: t[0],
			tokens: this.lexer.inline(t[0])
		};
	}
	escape(e) {
		let t = this.rules.inline.escape.exec(e);
		if (t) return {
			type: "escape",
			raw: t[0],
			text: t[1]
		};
	}
	tag(e) {
		let t = this.rules.inline.tag.exec(e);
		if (t) return !this.lexer.state.inLink && this.rules.other.startATag.test(t[0]) ? this.lexer.state.inLink = !0 : this.lexer.state.inLink && this.rules.other.endATag.test(t[0]) && (this.lexer.state.inLink = !1), !this.lexer.state.inRawBlock && this.rules.other.startPreScriptTag.test(t[0]) ? this.lexer.state.inRawBlock = !0 : this.lexer.state.inRawBlock && this.rules.other.endPreScriptTag.test(t[0]) && (this.lexer.state.inRawBlock = !1), {
			type: "html",
			raw: t[0],
			inLink: this.lexer.state.inLink,
			inRawBlock: this.lexer.state.inRawBlock,
			block: !1,
			text: t[0]
		};
	}
	link(e) {
		let t = this.rules.inline.link.exec(e);
		if (t) {
			let e = t[2].trim();
			if (!this.options.pedantic && this.rules.other.startAngleBracket.test(e)) {
				if (!this.rules.other.endAngleBracket.test(e)) return;
				let t = J(e.slice(0, -1), "\\");
				if ((e.length - t.length) % 2 == 0) return;
			} else {
				let e = Ze(t[2], "()");
				if (e === -2) return;
				if (e > -1) {
					let n = (t[0].indexOf("!") === 0 ? 5 : 4) + t[1].length + e;
					t[2] = t[2].substring(0, e), t[0] = t[0].substring(0, n).trim(), t[3] = "";
				}
			}
			let n = t[2], r = "";
			if (this.options.pedantic) {
				let e = this.rules.other.pedanticHrefTitle.exec(n);
				e && (n = e[1], r = e[3]);
			} else r = t[3] ? t[3].slice(1, -1) : "";
			return n = n.trim(), this.rules.other.startAngleBracket.test(n) && (n = this.options.pedantic && !this.rules.other.endAngleBracket.test(e) ? n.slice(1) : n.slice(1, -1)), $e(t, {
				href: n && n.replace(this.rules.inline.anyPunctuation, "$1"),
				title: r && r.replace(this.rules.inline.anyPunctuation, "$1")
			}, t[0], this.lexer, this.rules);
		}
	}
	reflink(e, t) {
		let n;
		if ((n = this.rules.inline.reflink.exec(e)) || (n = this.rules.inline.nolink.exec(e))) {
			let e = t[(n[2] || n[1]).replace(this.rules.other.multipleSpaceGlobal, " ").toLowerCase()];
			if (!e) {
				let e = n[0].charAt(0);
				return {
					type: "text",
					raw: e,
					text: e
				};
			}
			return $e(n, e, n[0], this.lexer, this.rules);
		}
	}
	emStrong(e, t, n = "") {
		let r = this.rules.inline.emStrongLDelim.exec(e);
		if (!(!r || !r[1] && !r[2] && !r[3] && !r[4] || r[4] && n.match(this.rules.other.unicodeAlphaNumeric)) && (!(r[1] || r[3]) || !n || this.rules.inline.punctuation.exec(n))) {
			let n = [...r[0]].length - 1, i, a, o = n, s = 0, c = r[0][0] === "*" ? this.rules.inline.emStrongRDelimAst : this.rules.inline.emStrongRDelimUnd;
			for (c.lastIndex = 0, t = t.slice(-1 * e.length + n); (r = c.exec(t)) != null;) {
				if (i = r[1] || r[2] || r[3] || r[4] || r[5] || r[6], !i) continue;
				if (a = [...i].length, r[3] || r[4]) {
					o += a;
					continue;
				} else if ((r[5] || r[6]) && n % 3 && !((n + a) % 3)) {
					s += a;
					continue;
				}
				if (o -= a, o > 0) continue;
				a = Math.min(a, a + o + s);
				let t = [...r[0]][0].length, c = e.slice(0, n + r.index + t + a);
				if (Math.min(n, a) % 2) {
					let e = c.slice(1, -1);
					return {
						type: "em",
						raw: c,
						text: e,
						tokens: this.lexer.inlineTokens(e)
					};
				}
				let l = c.slice(2, -2);
				return {
					type: "strong",
					raw: c,
					text: l,
					tokens: this.lexer.inlineTokens(l)
				};
			}
		}
	}
	codespan(e) {
		let t = this.rules.inline.code.exec(e);
		if (t) {
			let e = t[2].replace(this.rules.other.newLineCharGlobal, " "), n = this.rules.other.nonSpaceChar.test(e), r = this.rules.other.startingSpaceChar.test(e) && this.rules.other.endingSpaceChar.test(e);
			return n && r && (e = e.substring(1, e.length - 1)), {
				type: "codespan",
				raw: t[0],
				text: e
			};
		}
	}
	br(e) {
		let t = this.rules.inline.br.exec(e);
		if (t) return {
			type: "br",
			raw: t[0]
		};
	}
	del(e, t, n = "") {
		let r = this.rules.inline.delLDelim.exec(e);
		if (r && (!r[1] || !n || this.rules.inline.punctuation.exec(n))) {
			let n = [...r[0]].length - 1, i, a, o = n, s = this.rules.inline.delRDelim;
			for (s.lastIndex = 0, t = t.slice(-1 * e.length + n); (r = s.exec(t)) != null;) {
				if (i = r[1] || r[2] || r[3] || r[4] || r[5] || r[6], !i || (a = [...i].length, a !== n)) continue;
				if (r[3] || r[4]) {
					o += a;
					continue;
				}
				if (o -= a, o > 0) continue;
				a = Math.min(a, a + o);
				let t = [...r[0]][0].length, s = e.slice(0, n + r.index + t + a), c = s.slice(n, -n);
				return {
					type: "del",
					raw: s,
					text: c,
					tokens: this.lexer.inlineTokens(c)
				};
			}
		}
	}
	autolink(e) {
		let t = this.rules.inline.autolink.exec(e);
		if (t) {
			let e, n;
			return t[2] === "@" ? (e = t[1], n = "mailto:" + e) : (e = t[1], n = e), {
				type: "link",
				raw: t[0],
				text: e,
				href: n,
				tokens: [{
					type: "text",
					raw: e,
					text: e
				}]
			};
		}
	}
	url(e) {
		let t;
		if (t = this.rules.inline.url.exec(e)) {
			let e, n;
			if (t[2] === "@") e = t[0], n = "mailto:" + e;
			else {
				let r;
				do
					r = t[0], t[0] = this.rules.inline._backpedal.exec(t[0])?.[0] ?? "";
				while (r !== t[0]);
				e = t[0], n = t[1] === "www." ? "http://" + t[0] : t[0];
			}
			return {
				type: "link",
				raw: t[0],
				text: e,
				href: n,
				tokens: [{
					type: "text",
					raw: e,
					text: e
				}]
			};
		}
	}
	inlineText(e) {
		let t = this.rules.inline.text.exec(e);
		if (t) {
			let e = this.lexer.state.inRawBlock;
			return {
				type: "text",
				raw: t[0],
				text: t[0],
				escaped: e
			};
		}
	}
}, Y = class e {
	tokens;
	options;
	state;
	inlineQueue;
	tokenizer;
	constructor(e) {
		this.tokens = [], this.tokens.links = Object.create(null), this.options = e || E, this.options.tokenizer = this.options.tokenizer || new tt(), this.tokenizer = this.options.tokenizer, this.tokenizer.options = this.options, this.tokenizer.lexer = this, this.inlineQueue = [], this.state = {
			inLink: !1,
			inRawBlock: !1,
			top: !0
		};
		let t = {
			other: A,
			block: Ke.normal,
			inline: K.normal
		};
		this.options.pedantic ? (t.block = Ke.pedantic, t.inline = K.pedantic) : this.options.gfm && (t.block = Ke.gfm, this.options.breaks ? t.inline = K.breaks : t.inline = K.gfm), this.tokenizer.rules = t;
	}
	static get rules() {
		return {
			block: Ke,
			inline: K
		};
	}
	static lex(t, n) {
		return new e(n).lex(t);
	}
	static lexInline(t, n) {
		return new e(n).inlineTokens(t);
	}
	lex(e) {
		e = e.replace(A.carriageReturn, "\n"), this.blockTokens(e, this.tokens);
		for (let e = 0; e < this.inlineQueue.length; e++) {
			let t = this.inlineQueue[e];
			this.inlineTokens(t.src, t.tokens);
		}
		return this.inlineQueue = [], this.tokens;
	}
	blockTokens(e, t = [], n = !1) {
		for (this.tokenizer.lexer = this, this.options.pedantic && (e = e.replace(A.tabCharGlobal, "    ").replace(A.spaceLine, "")); e;) {
			let r;
			if (this.options.extensions?.block?.some((n) => (r = n.call({ lexer: this }, e, t)) ? (e = e.substring(r.raw.length), t.push(r), !0) : !1)) continue;
			if (r = this.tokenizer.space(e)) {
				e = e.substring(r.raw.length);
				let n = t.at(-1);
				r.raw.length === 1 && n !== void 0 ? n.raw += "\n" : t.push(r);
				continue;
			}
			if (r = this.tokenizer.code(e)) {
				e = e.substring(r.raw.length);
				let n = t.at(-1);
				n?.type === "paragraph" || n?.type === "text" ? (n.raw += (n.raw.endsWith("\n") ? "" : "\n") + r.raw, n.text += "\n" + r.text, this.inlineQueue.at(-1).src = n.text) : t.push(r);
				continue;
			}
			if (r = this.tokenizer.fences(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.heading(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.hr(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.blockquote(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.list(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.html(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.def(e)) {
				e = e.substring(r.raw.length);
				let n = t.at(-1);
				n?.type === "paragraph" || n?.type === "text" ? (n.raw += (n.raw.endsWith("\n") ? "" : "\n") + r.raw, n.text += "\n" + r.raw, this.inlineQueue.at(-1).src = n.text) : this.tokens.links[r.tag] || (this.tokens.links[r.tag] = {
					href: r.href,
					title: r.title
				}, t.push(r));
				continue;
			}
			if (r = this.tokenizer.table(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.lheading(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			let i = e;
			if (this.options.extensions?.startBlock) {
				let t = Infinity, n = e.slice(1), r;
				this.options.extensions.startBlock.forEach((e) => {
					r = e.call({ lexer: this }, n), typeof r == "number" && r >= 0 && (t = Math.min(t, r));
				}), t < Infinity && t >= 0 && (i = e.substring(0, t + 1));
			}
			if (this.state.top && (r = this.tokenizer.paragraph(i))) {
				let a = t.at(-1);
				n && a?.type === "paragraph" ? (a.raw += (a.raw.endsWith("\n") ? "" : "\n") + r.raw, a.text += "\n" + r.text, this.inlineQueue.pop(), this.inlineQueue.at(-1).src = a.text) : t.push(r), n = i.length !== e.length, e = e.substring(r.raw.length);
				continue;
			}
			if (r = this.tokenizer.text(e)) {
				e = e.substring(r.raw.length);
				let n = t.at(-1);
				n?.type === "text" ? (n.raw += (n.raw.endsWith("\n") ? "" : "\n") + r.raw, n.text += "\n" + r.text, this.inlineQueue.pop(), this.inlineQueue.at(-1).src = n.text) : t.push(r);
				continue;
			}
			if (e) {
				let t = "Infinite loop on byte: " + e.charCodeAt(0);
				if (this.options.silent) {
					console.error(t);
					break;
				} else throw Error(t);
			}
		}
		return this.state.top = !0, t;
	}
	inline(e, t = []) {
		return this.inlineQueue.push({
			src: e,
			tokens: t
		}), t;
	}
	inlineTokens(e, t = []) {
		this.tokenizer.lexer = this;
		let n = e, r = null;
		if (this.tokens.links) {
			let e = Object.keys(this.tokens.links);
			if (e.length > 0) for (; (r = this.tokenizer.rules.inline.reflinkSearch.exec(n)) != null;) e.includes(r[0].slice(r[0].lastIndexOf("[") + 1, -1)) && (n = n.slice(0, r.index) + "[" + "a".repeat(r[0].length - 2) + "]" + n.slice(this.tokenizer.rules.inline.reflinkSearch.lastIndex));
		}
		for (; (r = this.tokenizer.rules.inline.anyPunctuation.exec(n)) != null;) n = n.slice(0, r.index) + "++" + n.slice(this.tokenizer.rules.inline.anyPunctuation.lastIndex);
		let i;
		for (; (r = this.tokenizer.rules.inline.blockSkip.exec(n)) != null;) i = r[2] ? r[2].length : 0, n = n.slice(0, r.index + i) + "[" + "a".repeat(r[0].length - i - 2) + "]" + n.slice(this.tokenizer.rules.inline.blockSkip.lastIndex);
		n = this.options.hooks?.emStrongMask?.call({ lexer: this }, n) ?? n;
		let a = !1, o = "";
		for (; e;) {
			a || (o = ""), a = !1;
			let r;
			if (this.options.extensions?.inline?.some((n) => (r = n.call({ lexer: this }, e, t)) ? (e = e.substring(r.raw.length), t.push(r), !0) : !1)) continue;
			if (r = this.tokenizer.escape(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.tag(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.link(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.reflink(e, this.tokens.links)) {
				e = e.substring(r.raw.length);
				let n = t.at(-1);
				r.type === "text" && n?.type === "text" ? (n.raw += r.raw, n.text += r.text) : t.push(r);
				continue;
			}
			if (r = this.tokenizer.emStrong(e, n, o)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.codespan(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.br(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.del(e, n, o)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (r = this.tokenizer.autolink(e)) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			if (!this.state.inLink && (r = this.tokenizer.url(e))) {
				e = e.substring(r.raw.length), t.push(r);
				continue;
			}
			let i = e;
			if (this.options.extensions?.startInline) {
				let t = Infinity, n = e.slice(1), r;
				this.options.extensions.startInline.forEach((e) => {
					r = e.call({ lexer: this }, n), typeof r == "number" && r >= 0 && (t = Math.min(t, r));
				}), t < Infinity && t >= 0 && (i = e.substring(0, t + 1));
			}
			if (r = this.tokenizer.inlineText(i)) {
				e = e.substring(r.raw.length), r.raw.slice(-1) !== "_" && (o = r.raw.slice(-1)), a = !0;
				let n = t.at(-1);
				n?.type === "text" ? (n.raw += r.raw, n.text += r.text) : t.push(r);
				continue;
			}
			if (e) {
				let t = "Infinite loop on byte: " + e.charCodeAt(0);
				if (this.options.silent) {
					console.error(t);
					break;
				} else throw Error(t);
			}
		}
		return t;
	}
}, nt = class {
	options;
	parser;
	constructor(e) {
		this.options = e || E;
	}
	space(e) {
		return "";
	}
	code({ text: e, lang: t, escaped: n }) {
		let r = (t || "").match(A.notSpaceStart)?.[0], i = e.replace(A.endingNewline, "") + "\n";
		return r ? "<pre><code class=\"language-" + q(r) + "\">" + (n ? i : q(i, !0)) + "</code></pre>\n" : "<pre><code>" + (n ? i : q(i, !0)) + "</code></pre>\n";
	}
	blockquote({ tokens: e }) {
		return `<blockquote>
${this.parser.parse(e)}</blockquote>
`;
	}
	html({ text: e }) {
		return e;
	}
	def(e) {
		return "";
	}
	heading({ tokens: e, depth: t }) {
		return `<h${t}>${this.parser.parseInline(e)}</h${t}>
`;
	}
	hr(e) {
		return "<hr>\n";
	}
	list(e) {
		let t = e.ordered, n = e.start, r = "";
		for (let t = 0; t < e.items.length; t++) {
			let n = e.items[t];
			r += this.listitem(n);
		}
		let i = t ? "ol" : "ul", a = t && n !== 1 ? " start=\"" + n + "\"" : "";
		return "<" + i + a + ">\n" + r + "</" + i + ">\n";
	}
	listitem(e) {
		return `<li>${this.parser.parse(e.tokens)}</li>
`;
	}
	checkbox({ checked: e }) {
		return "<input " + (e ? "checked=\"\" " : "") + "disabled=\"\" type=\"checkbox\"> ";
	}
	paragraph({ tokens: e }) {
		return `<p>${this.parser.parseInline(e)}</p>
`;
	}
	table(e) {
		let t = "", n = "";
		for (let t = 0; t < e.header.length; t++) n += this.tablecell(e.header[t]);
		t += this.tablerow({ text: n });
		let r = "";
		for (let t = 0; t < e.rows.length; t++) {
			let i = e.rows[t];
			n = "";
			for (let e = 0; e < i.length; e++) n += this.tablecell(i[e]);
			r += this.tablerow({ text: n });
		}
		return r &&= `<tbody>${r}</tbody>`, "<table>\n<thead>\n" + t + "</thead>\n" + r + "</table>\n";
	}
	tablerow({ text: e }) {
		return `<tr>
${e}</tr>
`;
	}
	tablecell(e) {
		let t = this.parser.parseInline(e.tokens), n = e.header ? "th" : "td";
		return (e.align ? `<${n} align="${e.align}">` : `<${n}>`) + t + `</${n}>
`;
	}
	strong({ tokens: e }) {
		return `<strong>${this.parser.parseInline(e)}</strong>`;
	}
	em({ tokens: e }) {
		return `<em>${this.parser.parseInline(e)}</em>`;
	}
	codespan({ text: e }) {
		return `<code>${q(e, !0)}</code>`;
	}
	br(e) {
		return "<br>";
	}
	del({ tokens: e }) {
		return `<del>${this.parser.parseInline(e)}</del>`;
	}
	link({ href: e, title: t, tokens: n }) {
		let r = this.parser.parseInline(n), i = Ye(e);
		if (i === null) return r;
		e = i;
		let a = "<a href=\"" + e + "\"";
		return t && (a += " title=\"" + q(t) + "\""), a += ">" + r + "</a>", a;
	}
	image({ href: e, title: t, text: n, tokens: r }) {
		r && (n = this.parser.parseInline(r, this.parser.textRenderer));
		let i = Ye(e);
		if (i === null) return q(n);
		e = i;
		let a = `<img src="${e}" alt="${q(n)}"`;
		return t && (a += ` title="${q(t)}"`), a += ">", a;
	}
	text(e) {
		return "tokens" in e && e.tokens ? this.parser.parseInline(e.tokens) : "escaped" in e && e.escaped ? e.text : q(e.text);
	}
}, rt = class {
	strong({ text: e }) {
		return e;
	}
	em({ text: e }) {
		return e;
	}
	codespan({ text: e }) {
		return e;
	}
	del({ text: e }) {
		return e;
	}
	html({ text: e }) {
		return e;
	}
	text({ text: e }) {
		return e;
	}
	link({ text: e }) {
		return "" + e;
	}
	image({ text: e }) {
		return "" + e;
	}
	br() {
		return "";
	}
	checkbox({ raw: e }) {
		return e;
	}
}, X = class e {
	options;
	renderer;
	textRenderer;
	constructor(e) {
		this.options = e || E, this.options.renderer = this.options.renderer || new nt(), this.renderer = this.options.renderer, this.renderer.options = this.options, this.renderer.parser = this, this.textRenderer = new rt();
	}
	static parse(t, n) {
		return new e(n).parse(t);
	}
	static parseInline(t, n) {
		return new e(n).parseInline(t);
	}
	parse(e) {
		this.renderer.parser = this;
		let t = "";
		for (let n = 0; n < e.length; n++) {
			let r = e[n];
			if (this.options.extensions?.renderers?.[r.type]) {
				let e = r, n = this.options.extensions.renderers[e.type].call({ parser: this }, e);
				if (n !== !1 || ![
					"space",
					"hr",
					"heading",
					"code",
					"table",
					"blockquote",
					"list",
					"html",
					"def",
					"paragraph",
					"text"
				].includes(e.type)) {
					t += n || "";
					continue;
				}
			}
			let i = r;
			switch (i.type) {
				case "space":
					t += this.renderer.space(i);
					break;
				case "hr":
					t += this.renderer.hr(i);
					break;
				case "heading":
					t += this.renderer.heading(i);
					break;
				case "code":
					t += this.renderer.code(i);
					break;
				case "table":
					t += this.renderer.table(i);
					break;
				case "blockquote":
					t += this.renderer.blockquote(i);
					break;
				case "list":
					t += this.renderer.list(i);
					break;
				case "checkbox":
					t += this.renderer.checkbox(i);
					break;
				case "html":
					t += this.renderer.html(i);
					break;
				case "def":
					t += this.renderer.def(i);
					break;
				case "paragraph":
					t += this.renderer.paragraph(i);
					break;
				case "text":
					t += this.renderer.text(i);
					break;
				default: {
					let e = "Token with \"" + i.type + "\" type was not found.";
					if (this.options.silent) return console.error(e), "";
					throw Error(e);
				}
			}
		}
		return t;
	}
	parseInline(e, t = this.renderer) {
		this.renderer.parser = this;
		let n = "";
		for (let r = 0; r < e.length; r++) {
			let i = e[r];
			if (this.options.extensions?.renderers?.[i.type]) {
				let e = this.options.extensions.renderers[i.type].call({ parser: this }, i);
				if (e !== !1 || ![
					"escape",
					"html",
					"link",
					"image",
					"strong",
					"em",
					"codespan",
					"br",
					"del",
					"text"
				].includes(i.type)) {
					n += e || "";
					continue;
				}
			}
			let a = i;
			switch (a.type) {
				case "escape":
					n += t.text(a);
					break;
				case "html":
					n += t.html(a);
					break;
				case "link":
					n += t.link(a);
					break;
				case "image":
					n += t.image(a);
					break;
				case "checkbox":
					n += t.checkbox(a);
					break;
				case "strong":
					n += t.strong(a);
					break;
				case "em":
					n += t.em(a);
					break;
				case "codespan":
					n += t.codespan(a);
					break;
				case "br":
					n += t.br(a);
					break;
				case "del":
					n += t.del(a);
					break;
				case "text":
					n += t.text(a);
					break;
				default: {
					let e = "Token with \"" + a.type + "\" type was not found.";
					if (this.options.silent) return console.error(e), "";
					throw Error(e);
				}
			}
		}
		return n;
	}
}, Z = class {
	options;
	block;
	constructor(e) {
		this.options = e || E;
	}
	static passThroughHooks = new Set([
		"preprocess",
		"postprocess",
		"processAllTokens",
		"emStrongMask"
	]);
	static passThroughHooksRespectAsync = new Set([
		"preprocess",
		"postprocess",
		"processAllTokens"
	]);
	preprocess(e) {
		return e;
	}
	postprocess(e) {
		return e;
	}
	processAllTokens(e) {
		return e;
	}
	emStrongMask(e) {
		return e;
	}
	provideLexer() {
		return this.block ? Y.lex : Y.lexInline;
	}
	provideParser() {
		return this.block ? X.parse : X.parseInline;
	}
}, it = class {
	defaults = T();
	options = this.setOptions;
	parse = this.parseMarkdown(!0);
	parseInline = this.parseMarkdown(!1);
	Parser = X;
	Renderer = nt;
	TextRenderer = rt;
	Lexer = Y;
	Tokenizer = tt;
	Hooks = Z;
	constructor(...e) {
		this.use(...e);
	}
	walkTokens(e, t) {
		let n = [];
		for (let r of e) switch (n = n.concat(t.call(this, r)), r.type) {
			case "table": {
				let e = r;
				for (let r of e.header) n = n.concat(this.walkTokens(r.tokens, t));
				for (let r of e.rows) for (let e of r) n = n.concat(this.walkTokens(e.tokens, t));
				break;
			}
			case "list": {
				let e = r;
				n = n.concat(this.walkTokens(e.items, t));
				break;
			}
			default: {
				let e = r;
				this.defaults.extensions?.childTokens?.[e.type] ? this.defaults.extensions.childTokens[e.type].forEach((r) => {
					let i = e[r].flat(Infinity);
					n = n.concat(this.walkTokens(i, t));
				}) : e.tokens && (n = n.concat(this.walkTokens(e.tokens, t)));
			}
		}
		return n;
	}
	use(...e) {
		let t = this.defaults.extensions || {
			renderers: {},
			childTokens: {}
		};
		return e.forEach((e) => {
			let n = { ...e };
			if (n.async = this.defaults.async || n.async || !1, e.extensions && (e.extensions.forEach((e) => {
				if (!e.name) throw Error("extension name required");
				if ("renderer" in e) {
					let n = t.renderers[e.name];
					n ? t.renderers[e.name] = function(...t) {
						let r = e.renderer.apply(this, t);
						return r === !1 && (r = n.apply(this, t)), r;
					} : t.renderers[e.name] = e.renderer;
				}
				if ("tokenizer" in e) {
					if (!e.level || e.level !== "block" && e.level !== "inline") throw Error("extension level must be 'block' or 'inline'");
					let n = t[e.level];
					n ? n.unshift(e.tokenizer) : t[e.level] = [e.tokenizer], e.start && (e.level === "block" ? t.startBlock ? t.startBlock.push(e.start) : t.startBlock = [e.start] : e.level === "inline" && (t.startInline ? t.startInline.push(e.start) : t.startInline = [e.start]));
				}
				"childTokens" in e && e.childTokens && (t.childTokens[e.name] = e.childTokens);
			}), n.extensions = t), e.renderer) {
				let t = this.defaults.renderer || new nt(this.defaults);
				for (let n in e.renderer) {
					if (!(n in t)) throw Error(`renderer '${n}' does not exist`);
					if (["options", "parser"].includes(n)) continue;
					let r = n, i = e.renderer[r], a = t[r];
					t[r] = (...e) => {
						let n = i.apply(t, e);
						return n === !1 && (n = a.apply(t, e)), n || "";
					};
				}
				n.renderer = t;
			}
			if (e.tokenizer) {
				let t = this.defaults.tokenizer || new tt(this.defaults);
				for (let n in e.tokenizer) {
					if (!(n in t)) throw Error(`tokenizer '${n}' does not exist`);
					if ([
						"options",
						"rules",
						"lexer"
					].includes(n)) continue;
					let r = n, i = e.tokenizer[r], a = t[r];
					t[r] = (...e) => {
						let n = i.apply(t, e);
						return n === !1 && (n = a.apply(t, e)), n;
					};
				}
				n.tokenizer = t;
			}
			if (e.hooks) {
				let t = this.defaults.hooks || new Z();
				for (let n in e.hooks) {
					if (!(n in t)) throw Error(`hook '${n}' does not exist`);
					if (["options", "block"].includes(n)) continue;
					let r = n, i = e.hooks[r], a = t[r];
					Z.passThroughHooks.has(n) ? t[r] = (e) => {
						if (this.defaults.async && Z.passThroughHooksRespectAsync.has(n)) return (async () => {
							let n = await i.call(t, e);
							return a.call(t, n);
						})();
						let r = i.call(t, e);
						return a.call(t, r);
					} : t[r] = (...e) => {
						if (this.defaults.async) return (async () => {
							let n = await i.apply(t, e);
							return n === !1 && (n = await a.apply(t, e)), n;
						})();
						let n = i.apply(t, e);
						return n === !1 && (n = a.apply(t, e)), n;
					};
				}
				n.hooks = t;
			}
			if (e.walkTokens) {
				let t = this.defaults.walkTokens, r = e.walkTokens;
				n.walkTokens = function(e) {
					let n = [];
					return n.push(r.call(this, e)), t && (n = n.concat(t.call(this, e))), n;
				};
			}
			this.defaults = {
				...this.defaults,
				...n
			};
		}), this;
	}
	setOptions(e) {
		return this.defaults = {
			...this.defaults,
			...e
		}, this;
	}
	lexer(e, t) {
		return Y.lex(e, t ?? this.defaults);
	}
	parser(e, t) {
		return X.parse(e, t ?? this.defaults);
	}
	parseMarkdown(e) {
		return (t, n) => {
			let r = { ...n }, i = {
				...this.defaults,
				...r
			}, a = this.onError(!!i.silent, !!i.async);
			if (this.defaults.async === !0 && r.async === !1) return a(/* @__PURE__ */ Error("marked(): The async option was set to true by an extension. Remove async: false from the parse options object to return a Promise."));
			if (typeof t > "u" || t === null) return a(/* @__PURE__ */ Error("marked(): input parameter is undefined or null"));
			if (typeof t != "string") return a(/* @__PURE__ */ Error("marked(): input parameter is of type " + Object.prototype.toString.call(t) + ", string expected"));
			if (i.hooks && (i.hooks.options = i, i.hooks.block = e), i.async) return (async () => {
				let n = i.hooks ? await i.hooks.preprocess(t) : t, r = await (i.hooks ? await i.hooks.provideLexer() : e ? Y.lex : Y.lexInline)(n, i), a = i.hooks ? await i.hooks.processAllTokens(r) : r;
				i.walkTokens && await Promise.all(this.walkTokens(a, i.walkTokens));
				let o = await (i.hooks ? await i.hooks.provideParser() : e ? X.parse : X.parseInline)(a, i);
				return i.hooks ? await i.hooks.postprocess(o) : o;
			})().catch(a);
			try {
				i.hooks && (t = i.hooks.preprocess(t));
				let n = (i.hooks ? i.hooks.provideLexer() : e ? Y.lex : Y.lexInline)(t, i);
				i.hooks && (n = i.hooks.processAllTokens(n)), i.walkTokens && this.walkTokens(n, i.walkTokens);
				let r = (i.hooks ? i.hooks.provideParser() : e ? X.parse : X.parseInline)(n, i);
				return i.hooks && (r = i.hooks.postprocess(r)), r;
			} catch (e) {
				return a(e);
			}
		};
	}
	onError(e, t) {
		return (n) => {
			if (n.message += "\nPlease report this to https://github.com/markedjs/marked.", e) {
				let e = "<p>An error occurred:</p><pre>" + q(n.message + "", !0) + "</pre>";
				return t ? Promise.resolve(e) : e;
			}
			if (t) return Promise.reject(n);
			throw n;
		};
	}
}, Q = new it();
function $(e, t) {
	return Q.parse(e, t);
}
$.options = $.setOptions = function(e) {
	return Q.setOptions(e), $.defaults = Q.defaults, D($.defaults), $;
}, $.getDefaults = T, $.defaults = E, $.use = function(...e) {
	return Q.use(...e), $.defaults = Q.defaults, D($.defaults), $;
}, $.walkTokens = function(e, t) {
	return Q.walkTokens(e, t);
}, $.parseInline = Q.parseInline, $.Parser = X, $.parser = X.parse, $.Renderer = nt, $.TextRenderer = rt, $.Lexer = Y, $.lexer = Y.lex, $.Tokenizer = tt, $.Hooks = Z, $.parse = $, $.options, $.setOptions, $.use, $.walkTokens, $.parseInline, X.parse, Y.lex;
//#endregion
//#region ../node_modules/marked-smartypants/lib/index.mjs
var at = /<(\/?)(?:pre|code|kbd|script|math)[^>]*>/i, ot = (e = "", t = "1") => {
	var n, r, i, a, o, s = 0;
	if (t = typeof t == "number" ? t.toString() : t.replace(/\s/g, ""), t === "0") return e;
	if (t === "1") n = 1, r = 1, i = 1, a = 1;
	else if (t === "2") n = 1, r = 1, i = 2, a = 1;
	else if (t === "3") n = 1, r = 1, i = 3, a = 1;
	else if (t === "-1") o = 1;
	else for (let e = 0; e < t.length; e++) {
		let o = t[e];
		o === "q" && (n = 1), o === "b" && (r = 1), o === "B" && (r = 2), o === "d" && (i = 1), o === "D" && (i = 2), o === "i" && (i = 3), o === "e" && (a = 1), o === "w" && (s = 1);
	}
	var c = gt(e), l = "", u = 0, d = "";
	for (let e = 0; e < c.length; e++) {
		let t = c[e];
		if (t[0] === "tag") {
			l += t[1];
			let e = at.exec(t[1]);
			e && (u = e[1] === "/" ? 0 : 1);
		} else {
			let e = t[1], c = e.substring(e.length - 1, e.length);
			u || (e = ht(e), s && (e = e.replace(/$quot;/g, "\"")), i && (i === 1 && (e = ut(e)), i === 2 && (e = dt(e)), i === 3 && (e = ft(e))), a && (e = pt(e)), r && (e = ct(e), r === 2 && (e = lt(e))), n && (e = e === "'" ? /\S/.test(d) ? "&#8217;" : "&#8216;" : e === "\"" ? /\S/.test(d) ? "&#8221;" : "&#8220;" : st(e)), o && (e = mt(e))), d = c, l += e;
		}
	}
	return l;
}, st = (e) => {
	var t = "[!\"#$%'()*+,-./:;<=>?@[\\]^_`{|}~]";
	e = e.replace(RegExp(`^'(?=${t}\\B)`), "&#8217;"), e = e.replace(RegExp(`^"(?=${t}\\B)`), "&#8221;"), e = e.replace(/"'(?=\w)/, "&#8220;&#8216;"), e = e.replace(/'"(?=\w)/, "&#8216;&#8220;"), e = e.replace(/'(?=\d\d)/, "&#8217;");
	var n = "[^\\ \\t\\r\\n\\[\\{\\(\\-]", r = "[\\ \\t\\r\\n\\[\\{\\(\\-]", i = "&#8211;|&#8212;";
	return e = e.replace(RegExp(`(\\s|&nbsp;|--|&[mn]dash;|${i}|&#x201[34])'(?=\\w)`, "g"), "$1&#8216;"), e = e.replace(RegExp(`(${n})'`, "g"), "$1&#8217;"), e = e.replace(RegExp(`(${r}?)'(?=\\s|s\\b)`, "g"), "$1&#8217;"), e = e.replace(/'/g, "&#8216;"), e = e.replace(RegExp(`(\\s|&nbsp;|--|&[mn]dash;|${i}|&#x201[34])"(?=\\w)`, "g"), "$1&#8220;"), e = e.replace(RegExp(`(${n})"`, "g"), "$1&#8221;"), e = e.replace(RegExp(`(${r}?)"(?=\\s)`, "g"), "$1&#8221;"), e = e.replace(/"/g, "&#8220;"), e;
}, ct = (e) => (e = e.replace(/``/g, "&#8220;"), e = e.replace(/''/g, "&#8221;"), e), lt = (e) => (e = e.replace(/`/g, "&#8216;"), e = e.replace(/'/g, "&#8217;"), e), ut = (e) => (e = e.replace(/--/g, "&#8212;"), e), dt = (e) => (e = e.replace(/---/g, "&#8212;"), e = e.replace(/--/g, "&#8211;"), e), ft = (e) => (e = e.replace(/---/g, "&#8211;"), e = e.replace(/--/g, "&#8212;"), e), pt = (e) => (e = e.replace(/\.\.\./g, "&#8230;"), e = e.replace(/\. \. \./g, "&#8230;"), e), mt = (e) => (e = e.replace(/&#8211;/g, "-"), e = e.replace(/&#8212;/g, "--"), e = e.replace(/&#8216;/g, "'"), e = e.replace(/&#8217;/g, "'"), e = e.replace(/&#8220;/g, "\""), e = e.replace(/&#8221;/g, "\""), e = e.replace(/&#8230;/g, "..."), e), ht = (e) => (e = e.replace(/\\\\/g, "&#92;"), e = e.replace(/\\"/g, "&#34;"), e = e.replace(/\\'/g, "&#39;"), e = e.replace(/\\\./g, "&#46;"), e = e.replace(/\\-/g, "&#45;"), e = e.replace(/\\`/g, "&#96;"), e), gt = (e) => {
	for (var t = 0, n = e.length, r = [], i = /<!--[\s\S]*?-->|<\?.*?\?>|<[^>]*>/g, a = null; a = i.exec(e);) {
		if (t < a.index) {
			let n = ["text", e.substring(t, a.index)];
			r.push(n);
		}
		let n = ["tag", a.toString()];
		r.push(n), t = i.lastIndex;
	}
	if (t < n) {
		let i = ["text", e.substring(t, n)];
		r.push(i);
	}
	return r;
};
function _t({ config: e = 2 } = {}) {
	return {
		tokenizer: { inlineText(e) {
			let t = this.rules.inline.text.exec(e);
			if (t) return {
				type: "text",
				raw: t[0],
				text: t[0],
				escaped: !0
			};
		} },
		hooks: { postprocess(t) {
			return ot(t, e);
		} }
	};
}
//#endregion
//#region src/utils/markdown.ts
var vt = "\r?\n---\r?\n", yt = "^\\s*notes?:", bt = "\\.element\\s*?(.+?)$", xt = "\\.slide:\\s*?(\\S.+?)$", St = /\[\s*((\d*):)?\s*([\s\d,|-]*)\]/, Ct = {
	"&": "&amp;",
	"<": "&lt;",
	">": "&gt;",
	"\"": "&quot;",
	"'": "&#39;"
};
function wt(e) {
	let t = e.match(/^\n?(\s*)/)?.[1].length ?? 0, n = e.match(/^\n?(\t*)/)?.[1].length ?? 0;
	return n > 0 ? e.replace(RegExp(`\\n?\\t{${n}}(.*)`, "g"), (e, t) => `\n${t}`) : t > 1 ? e.replace(RegExp(`\\n? {${t}}(.*)`, "g"), (e, t) => `\n${t}`) : e;
}
function Tt(e) {
	return e.replace(/([&<>'"])/g, (e) => Ct[e]);
}
function Et(e = {}) {
	let { renderer: t, animateLists: n, smartypants: r, ...i } = e, a = t || { code({ text: e, lang: t }) {
		let n = t || "", r = "", i = "";
		if (St.test(n)) {
			let e = n.match(St)?.[2];
			e && (r = `data-ln-start-from="${e.trim()}"`), i = n.match(St)?.[3]?.trim() || "", i = `data-line-numbers="${i}"`, n = n.replace(St, "").trim();
		}
		return e = Tt(e), `<pre><code ${i} ${r} class="${n}">${e}</code></pre>`;
	} };
	n === !0 && !t && (a.listitem = function(e) {
		return `<li class="fragment">${e.tokens ? this.parser.parseInline(e.tokens) : e.text || ""}</li>`;
	});
	let o = new it();
	return o.use({
		renderer: a,
		...i
	}), r && o.use(_t()), o;
}
function Dt(e) {
	if (typeof e == "string") return e;
	throw Error("Async markdown parsing is not supported here because Reveal markdown parsing is synchronous.");
}
function Ot(e, t, n) {
	let r = e.split(new RegExp(n, "mgi")), i = e, a = "";
	return r.length === 2 && (i = r[0], a = `<aside class="notes">${Dt(t.parse(r[1].trim()))}</aside>`), `${Dt(t.parse(i))}${a}`;
}
function kt(e, t, n, r, i) {
	let a = RegExp(n + (r ? `|${r}` : ""), "mg"), o = new RegExp(n), s, c = 0, l = !0, u = !0, d = [];
	for (; s = a.exec(e);) {
		l = o.test(s[0]), !l && u && d.push([]);
		let t = e.substring(c, s.index);
		l && u ? d.push(t) : d[d.length - 1].push(t), c = a.lastIndex, u = l;
	}
	return (u ? d : d[d.length - 1]).push(e.substring(c)), d.map((e, n) => Array.isArray(e) ? {
		type: "stack",
		key: `h${n}`,
		slides: e.map((e, r) => ({
			key: `h${n}-v${r}`,
			html: Ot(e, t, i)
		}))
	} : {
		type: "slide",
		key: `h${n}`,
		slide: {
			key: `h${n}`,
			html: Ot(e, t, i)
		}
	});
}
function At(e, t, n) {
	if (!t || e.nodeType !== Node.COMMENT_NODE || e.nodeValue == null) return !1;
	let r = new RegExp(n, "mg"), i = /* @__PURE__ */ RegExp("([^\"= ]+?)=\"([^\"]+?)\"|(data-[^\"= ]+?)(?=[\" ])", "mg"), a = e.nodeValue, o = r.exec(a);
	if (!o) return !1;
	let s = o[1];
	a = a.substring(0, o.index) + a.substring(r.lastIndex), e.nodeValue = a;
	let c;
	for (; c = i.exec(s);) c[2] ? t.setAttribute(c[1], c[2]) : t.setAttribute(c[3], "");
	return !0;
}
function jt(e, t, n, r, i) {
	if ("childNodes" in t && t.childNodes.length > 0) {
		let a = t instanceof Element ? t : n;
		for (let n = 0; n < t.childNodes.length; n += 1) {
			let o = t.childNodes[n];
			if (n > 0) {
				let e = n - 1;
				for (; e >= 0;) {
					let n = t.childNodes[e];
					if (typeof n.setAttribute == "function" && n.tagName !== "BR") {
						a = n;
						break;
					}
					--e;
				}
			}
			let s = e;
			o.nodeName === "SECTION" && (s = o, a = o), (typeof o.setAttribute == "function" || o.nodeType === Node.COMMENT_NODE) && jt(s, o, a, r, i);
		}
	}
	if (t.nodeType !== Node.COMMENT_NODE) return;
	let a = n;
	a && (a.tagName === "UL" || a.tagName === "OL") && (a = a.lastElementChild || a), At(t, a, r) === !1 && At(t, e, i);
}
function Mt(e) {
	let t = 5381;
	for (let n = 0; n < e.length; n += 1) t = t * 33 ^ e.charCodeAt(n);
	return (t >>> 0).toString(36);
}
function Nt(e) {
	return e == null || typeof e == "string" || typeof e == "number" || typeof e == "boolean" ? e : Array.isArray(e) ? e.map(Nt) : typeof e == "object" ? Object.entries(e).filter(([, e]) => e !== void 0 && typeof e != "function").sort(([e], [t]) => e.localeCompare(t)).map(([e, t]) => [e, Nt(t)]) : String(e);
}
function Pt(e) {
	return JSON.stringify(Object.entries(e).filter(([, e]) => e !== void 0 && typeof e != "function").sort(([e], [t]) => e.localeCompare(t)).map(([e, t]) => [e, Nt(t)]));
}
function Ft(e) {
	return e instanceof Error && e.message ? e.message : String(e);
}
//#endregion
//#region src/components/markdown.tsx
function It({ html: e, elementAttributesSeparator: t, slideAttributesSeparator: n, sectionProps: r }) {
	let i = a(h), o = l(null);
	return s(() => {
		let e = o.current;
		e && jt(e, e, null, t, n);
	}, [
		e,
		t,
		n
	]), s(() => {
		let e = o.current;
		if (!e) return;
		let t = i?.getPlugin?.("highlight")?.highlightBlock;
		if (typeof t != "function") return;
		let n = 0;
		Array.from(e.querySelectorAll("pre code")).forEach((e) => {
			e.getAttribute("data-highlighted") !== "yes" && (t(e), n += 1);
		});
		let r = i?.syncFragments;
		n > 0 && typeof r == "function" && r(e);
	}), /* @__PURE__ */ p("section", {
		ref: o,
		...r,
		dangerouslySetInnerHTML: { __html: e }
	});
}
function Lt({ children: e, markdown: t, src: n, charset: r, separator: i = vt, verticalSeparator: a = null, notesSeparator: s = yt, elementAttributesSeparator: c = bt, slideAttributesSeparator: l = xt, options: d, background: h, backgroundImage: ee, backgroundVideo: te, backgroundVideoLoop: g, backgroundVideoMuted: _, backgroundIframe: v, backgroundColor: y, backgroundGradient: b, backgroundSize: x, backgroundPosition: S, backgroundRepeat: w, backgroundOpacity: ne, backgroundTransition: T, visibility: E, autoAnimate: D, autoAnimateId: O, autoAnimateRestart: k, autoAnimateUnmatched: re, autoAnimateEasing: A, autoAnimateDuration: j, autoAnimateDelay: M, transition: N, transitionSpeed: P, autoSlide: F, notes: I, backgroundInteractive: L, preload: ie, ...ae }) {
	let [R, z] = u(null), [B, V] = u(null);
	o(() => {
		if (!n) {
			z(null), V(null);
			return;
		}
		let e = new AbortController();
		return z(null), V(null), (async () => {
			try {
				let t = await fetch(n, { signal: e.signal });
				if (!t.ok) throw Error(`HTTP status ${t.status}`);
				z(r ? new TextDecoder(r).decode(await t.arrayBuffer()) : await t.text());
			} catch (t) {
				if (e.signal.aborted) return;
				V(Ft(t));
			}
		})(), () => e.abort();
	}, [n, r]);
	let H = C(ae, {
		background: h,
		backgroundImage: ee,
		backgroundVideo: te,
		backgroundVideoLoop: g,
		backgroundVideoMuted: _,
		backgroundIframe: v,
		backgroundColor: y,
		backgroundGradient: b,
		backgroundSize: x,
		backgroundPosition: S,
		backgroundRepeat: w,
		backgroundOpacity: ne,
		backgroundTransition: T,
		visibility: E,
		autoAnimate: D,
		autoAnimateId: O,
		autoAnimateRestart: k,
		autoAnimateUnmatched: re,
		autoAnimateEasing: A,
		autoAnimateDuration: j,
		autoAnimateDelay: M,
		transition: N,
		transitionSpeed: P,
		autoSlide: F,
		notes: I,
		backgroundInteractive: L,
		preload: ie
	}), U = Mt(Pt(H));
	if (n && B) return /* @__PURE__ */ m("section", {
		"data-state": "alert",
		children: [`ERROR: The attempt to fetch ${n} failed. ${B}.`, /* @__PURE__ */ p("p", { children: "Remember that you need to serve the presentation from an HTTP server." })]
	});
	if (n && R == null) return null;
	let W = n ? R || "" : wt(t ?? e ?? "");
	try {
		let e = kt(W, Et(d), i, a, s), t = Mt(`${c}\0${l}`);
		return /* @__PURE__ */ p(f, { children: e.map((e) => e.type === "slide" ? /* @__PURE__ */ p(It, {
			html: e.slide.html,
			elementAttributesSeparator: c,
			slideAttributesSeparator: l,
			sectionProps: H
		}, `${e.key}-${U}-${t}-${Mt(e.slide.html)}`) : /* @__PURE__ */ p("section", {
			...H,
			children: e.slides.map((e) => /* @__PURE__ */ p(It, {
				html: e.html,
				elementAttributesSeparator: c,
				slideAttributesSeparator: l
			}, `${e.key}-${t}-${Mt(e.html)}`))
		}, `${e.key}-${U}`)) });
	} catch (e) {
		return /* @__PURE__ */ p("section", {
			"data-state": "alert",
			children: `ERROR: Failed to parse markdown. ${Ft(e)}.`
		});
	}
}
//#endregion
//#region src/components/fragment.tsx
function Rt(...e) {
	return e.filter(Boolean).join(" ");
}
function zt(e, t) {
	return e ? t ? {
		...e,
		...t
	} : e : t;
}
function Bt({ animation: r, index: a, as: o, asChild: s, className: c, style: l, children: u }) {
	let d = Rt("fragment", r, c);
	if (s) {
		let r;
		try {
			r = e.only(u);
		} catch {
			throw Error("Fragment with asChild expects exactly one React element child.");
		}
		if (!i(r) || r.type === t) throw Error("Fragment with asChild expects exactly one non-Fragment React element child.");
		let o = {
			className: Rt(r.props.className, d),
			style: zt(r.props.style, l)
		};
		return a !== void 0 && (o["data-fragment-index"] = a), n(r, o);
	}
	return /* @__PURE__ */ p(o ?? "span", {
		className: d,
		style: l,
		"data-fragment-index": a,
		children: u
	});
}
//#endregion
//#region src/components/code.tsx
function Vt(e) {
	let t = e.replace(/\r\n/g, "\n").split("\n");
	for (; t.length && t[0].trim().length === 0;) t.shift();
	for (; t.length && t[t.length - 1].trim().length === 0;) t.pop();
	if (!t.length) return "";
	let n = t.filter((e) => e.trim().length > 0).reduce((e, t) => Math.min(e, t.match(/^\s*/)?.[0].length ?? 0), Infinity);
	return t.map((e) => e.slice(n)).join("\n");
}
function Ht(e) {
	let t = e.parentElement;
	t && Array.from(t.children).forEach((t) => {
		t !== e && t instanceof HTMLElement && t.tagName === "CODE" && t.classList.contains("fragment") && t.remove();
	});
}
function Ut({ children: e, code: t, language: n, trim: r = !0, lineNumbers: i, startFrom: o, noEscape: u, codeClassName: d, codeStyle: f, codeProps: m, className: ee, style: te, ...g }) {
	let _ = a(h), v = l(null), y = l(""), b = typeof t == "string" ? t : typeof e == "string" ? e : "", x = c(() => r ? Vt(b) : b, [b, r]), S = i === !0 ? "" : i === !1 || i == null ? void 0 : String(i), C = [n, d].filter(Boolean).join(" "), w = ["code-wrapper", ee].filter(Boolean).join(" ");
	return s(() => {
		let e = v.current;
		if (!e || !_) return;
		let t = _.getPlugin?.("highlight");
		if (!t || typeof t.highlightBlock != "function") return;
		let r = [
			x,
			n || "",
			d || "",
			S == null ? "__none__" : `lineNumbers:${S}`,
			o == null ? "" : String(o),
			u ? "1" : "0"
		].join("::");
		if (y.current === r && e.getAttribute("data-highlighted") === "yes") return;
		Ht(e), e.textContent = x, e.removeAttribute("data-highlighted"), e.classList.remove("hljs"), e.classList.remove("has-highlights"), S == null ? e.removeAttribute("data-line-numbers") : e.setAttribute("data-line-numbers", S), o == null ? e.removeAttribute("data-ln-start-from") : e.setAttribute("data-ln-start-from", String(o)), u ? e.setAttribute("data-noescape", "") : e.removeAttribute("data-noescape"), t.highlightBlock(e);
		let i = typeof e.closest == "function" ? e.closest("section") : null;
		i && typeof _.syncFragments == "function" && _.syncFragments(i), y.current = r;
	}, [
		_,
		x,
		n,
		d,
		S,
		o,
		u
	]), /* @__PURE__ */ p("pre", {
		className: w,
		style: te,
		...g,
		children: /* @__PURE__ */ p("code", {
			...m,
			ref: v,
			className: C || void 0,
			style: f,
			"data-line-numbers": S,
			"data-ln-start-from": o,
			"data-noescape": u ? "" : void 0,
			children: x
		})
	});
}
//#endregion
//#region src/index.ts
function Wt() {
	return a(h);
}
//#endregion
export { Ut as Code, b as Deck, Bt as Fragment, Lt as Markdown, h as RevealContext, w as Slide, ne as Stack, Wt as useReveal };
